Sources/Basic/Lock.swift

```diff
@@ -58,10 +59,31 @@ public final class FileLock {
         self.cachePath = cachePath
     }
 
-    /// Try to aquire a lock. This method will block until lock the already aquired by other process.
+    /// Registers signal handler for SIGUSR1 and raises the signal
+    /// after the given number of seconds.
+    private func setupTimeout(seconds: Double) {
+        // Register signal handler to avoid process termination.
+        var action = sigaction()
+        action.__sigaction_u.__sa_handler = { _ in }
+        sigaction(SIGUSR1, &action, nil)
```

Comment:
How about having a SignalHandler global shared object where you can register to get callbacks for signals?